### PR TITLE
User-facing api docs: utils: plotting, slack, threading, validator, zmq_helpers

### PR DESCRIPTION
**Pull request by lakhotiaharshit**
Corrects user-facing API docs in utils. 
- plotting
- slack
- threading
- validator
- zmq_helpers

@mention one core developer
@astafan8

### CODE REVIEW

**Comment by jenshnielsen** [comment]
I fixed a bunch of other small issues: 

* Before Args: there should always be a blank line.
* Args: is a special section -> So you should not have stuff like ` Inherited Args (see validators.Ints)`
* Context: should be in an Args section

A few more things that could be fixed 

- [ ] The docstring for the baseclass here https://qcodes.github.io/Qcodes/api/utils/validators.html#qcodes.utils.validators.Validator does not render nicely
- [ ] Numbers https://qcodes.github.io/Qcodes/api/utils/validators.html#qcodes.utils.validators.Numbers is missing a intro part of the docstring

File: qcodes/utils/validators.py [hunk 1]
```diff
@@ -189,6 +192,8 @@ def __repr__(self) -> str:
 
 class Numbers(Validator):
     """
+    Requires a number  of type int, float, numpy.integer or numpy.floating.
+
     Args:
         min_value: Minimal value allowed, default -inf.
         max_value: Maximal value allowed, default inf.
```
Comment (jenshnielsen) [inline qcodes/utils/validators.py]: Could you fix the todo below too?

Comment (lakhotiaharshit) [reply qcodes/utils/validators.py]: OK

Comment (jenshnielsen) [reply qcodes/utils/validators.py]: Thanks :)

**Comment by jenshnielsen** [comment]
Could you also add a short docstring at the top of the module for validators.py 

This will be rendered here file:///C:/Users/jenielse/source/repos/Qcodes/docs/_build/html/api/utils/index.html on the right side of the table

File: qcodes/utils/validators.py (hunk 1 above)
Comment (lakhotiaharshit) [reply qcodes/utils/validators.py]: Fixed the docs to included `Raises` in all the methods.

**Comment by jenshnielsen** [comment]
Thanks this looks great

**Pull request merged**